gui/src/mainwindow.cpp

```diff
@@ -274,7 +274,13 @@ void MainWindow::ServerItemWidgetTriggered()
 	else
 	{
 		RegistDialog regist_dialog(settings, server.GetHostAddr(), this);
-		int r = regist_dialog.exec();
+		regist_dialog.show();
+
+		QEventLoop loop;
```

Comment:
Why?

Reply:
Without this loop the dialog closes immediately. It used to use dialog.exec() but this would lock focus and it would be impossible to view the login window as it was always behind the registration dialog. Pointing this out has actually  helped me to discover an issue with another change in this PR: `SettingsDialog::RegisterNewHost` I altered to use the show() logic but without the loop in oreder to allow switching focus to the WebView and the dialog never shows.

Do you have any suggestions for getting around this without loops? I tried using a lambda to remove the loop like below:

```		RegistDialog regist_dialog(settings, server.GetHostAddr(), this);
		connect(&regist_dialog, &QDialog::finished, this, [&regist_dialog, &server, this](int result) {
			if(result == QDialog::Accepted && !server.discovered) // success
			{
				ManualHost manual_host = server.manual_host;
				manual_host.Register(regist_dialog.GetRegisteredHost());
				settings->SetManualHost(manual_host);
			}
		});
		regist_dialog.show();
```

but without the loop itself the dialog just closed immediately.

Reply:
Ignore this. I worked out if i mage the web engine inside a Dialog I can take focus from an the exec dialog beneath